DFC.Digital/DFC.Digital.Web.Sitefinity.Core/DataEventHandling/MicroServicesPublishingPageBuilder.cs

```diff
@@ -34,44 +34,74 @@ public static IList<string> GetPageURLs(PageNode pageNode)
             return pageUrls;
         }
 
-        public MicroServicesPublishingPageData GetCompositePageForPageNode(string providerName, Type contentType, Guid itemId)
+        public MicroServicesPublishingPageData GetPublishedPage(Type contentType, Guid itemId, string providerName)
         {
-            var pageNode = sitefinityManagerProxy.GetPageNode(providerName, contentType, itemId);
-            var pageData = sitefinityManagerProxy.GetPageData(providerName, contentType, itemId);
+            var pageNode = sitefinityManagerProxy.GetPageNode(contentType, itemId, providerName);
+            var pageData = sitefinityManagerProxy.GetPageData(contentType, itemId, providerName);
+            return BuildPageData(pageNode, pageData, providerName);
+        }
 
-            var microServicesPublishingPageData = new MicroServicesPublishingPageData() { CanonicalName = sitefinityPageNodeProxy.GetPageName(pageNode).ToLower(), Id = itemId };
-            microServicesPublishingPageData.IncludeInSiteMap = pageNode.Crawlable;
-            microServicesPublishingPageData.BreadcrumbTitle = sitefinityPageDataProxy.GetTitle(pageData);
-            microServicesPublishingPageData.MetaTags = new MetaTags() { Description = sitefinityPageDataProxy.GetDescription(pageData), Keywords = sitefinityPageDataProxy.GetKeywords(pageData), Title = sitefinityPageDataProxy.GetHtmlTitle(pageData) };
-            microServicesPublishingPageData.Content = GetPageContentBlocks(providerName, contentType, itemId);
-            microServicesPublishingPageData.AlternativeNames = GetPageURLs(pageNode);
-            microServicesPublishingPageData.LastReviewed = sitefinityPageNodeProxy.GetLastPublishedDate(pageNode);
+        public MicroServicesPublishingPageData GetPreviewPage(string name)
+        {
+            var pageData = sitefinityManagerProxy.GetPageDataByName(name);
```

Comment:
can we change this way -
`return sitefinityManagerProxy.GetPageDataByName(name);`
instead of 
```
var pageData = sitefinityManagerProxy.GetPageDataByName(name);
if (pageData is null)
{
   return null;
}
```